ci/update_gomod.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/go-utils/log"
+	"github.com/solo-io/skv2/codegen/util"
+)
+
+// this file is a Go script which udpates the go.mod file to match the contents of Chart-template.yaml
+
+const (
+	glooMeshDependencyName          = "gloo-mesh"
+	glooMeshRepo                    = "github.com/solo-io/gloo-mesh"
+	glooMeshEnerpriseDependencyName = "gloo-mesh-extender"
+	glooMeshEnterpriseRepo          = "github.com/solo-io/gloo-mesh-enterprise"
+)
+
+var moduleRoot = util.GetModuleRoot()
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatalf("%v", err)
+	}
+}
+
+func run() error {
+	log.Printf("updating gomod to match chart-template...")
+	chartTemplateFile := filepath.Join(moduleRoot, "install", "helm", "gloo-mesh-enterprise", "Chart-template.yaml")
+
+	chartTemplate, err := ioutil.ReadFile(chartTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	var chartContents chartWithDependencies
+	if err := yaml.Unmarshal(chartTemplate, &chartContents); err != nil {
+		return err
+	}
+
+	var versions subchartVersions
+	for _, dep := range chartContents.Dependencies {
+		switch dep.Name {
+		case glooMeshDependencyName:
+			versions.glooMesh = dep.Version
+		case glooMeshEnerpriseDependencyName:
+			versions.glooMeshEnterprise = dep.Version
+		}
+	}
+
+	if versions.glooMesh == "" {
+		return eris.Errorf("no gloo-mesh version found")
+	}
+	if versions.glooMeshEnterprise == "" {
+		return eris.Errorf("no gloo-mesh-enterprise version found")
+	}
+
+	return versions.updateGoMod()
+}
+
+// the versions of subcharts we import
+type subchartVersions struct {
+	glooMesh           string
+	glooMeshEnterprise string
+}
+
+func (v subchartVersions) updateGoMod() error {
+	for repo, version := range map[string]string{
+		glooMeshRepo:           v.glooMesh,
+		glooMeshEnterpriseRepo: v.glooMeshEnterprise,
```

Comment:
Given that [enterprise's go mod declares OSS as a dependency](https://github.com/solo-io/gloo-mesh-enterprise/blob/master/go.mod#L36), would it be sufficient to just install the go mod deps for Gloo Mesh enterprise?

Reply:
we may want changes from OSS that do not require a version bump in enterprise. these e2e tests should verify compatibility